primitives/utils/src/mpsc.rs

```diff
@@ -246,6 +256,183 @@ mod inner {
 			Poll::Ready(Ok(()))
 		}
 	}
+
+	/// Wrapper Type around `Sender` that increases the global
+	/// measure when a message is added
+	#[derive(Debug)]
+	pub struct TracingSender<T>(&'static str, Sender<T>);
```

Comment:
Why do we need this?

Reply:
You said it wasn't right that the channel was unbounded. And I said okay but there is no implementation of the bounded version of this channel yet. So I made the implementation and that's the bounded channel.

What is your concern here exactly?
 - The fact that it's using a *tracing* channel instead of a `futures`' mpsc channel? I don't know, it was a tracing channel before so I suspect the author really wanted it to be tracing.
 - Why not re-using the sender of the unbounded channel? It seems the methods and signature are different for bounded and unbounded, so the type has to be different: https://docs.rs/futures/0.3.12/futures/channel/mpsc/struct.Sender.html and https://docs.rs/futures/0.3.12/futures/channel/mpsc/struct.UnboundedSender.html

Reply:
:exploding_head: 

Ohhh... I see :thinking: Makes sense all of a sudden :grin:

Reply:
Done in 9dae7e4ec818ea0382e6306545c71cd84423d893 (and 1b5df60760c95ebf522aa62d3ccbc6338c42bc96)